File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.py
class Solution(object):
    def longestOnes(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        i = 0
        max_len = 0
        zero_count = 0
        
        for j in range(len(nums)):
            if nums[j] == 0:
                zero_count += 1
                
            while zero_count > k:
                if nums[i] == 0:
                    zero_count -= 1
                i += 1
                
            max_len = max(max_len, j - i + 1)

        return max_len

File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.rs
impl Solution {
    pub fn longest_ones(nums: Vec<i32>, k: i32) -> i32 {
        let(mut i, mut zero_cnt, mut max_len) = (0, 0, 0);
            for j in 0..nums.len() {
                if nums[j] == 0{
                    zero_cnt += 1;
                }
            
                while zero_cnt > k {
                   if nums[i as usize] == 0 {
                       zero_cnt -= 1;
                    }
                    i += 1;
                }
                max_len = max_len.max((j as i32) - i + 1);
                
            }
            max_len
//         let (mut i, mut max_len, mut zero_cnt) = (0, 0, 0);
//         for j in 0..nums.len() {
//             if nums[j] == 0 {
//                 zero_cnt += 1;
//             }
        
//             while zero_cnt > k {
//                 if nums[i as usize] == 0 {
//                     zero_cnt -= 1;
//                 }
//                 i += 1;
//             }
//             max_len = max_len.max((j as i32) - i + 1);
//         }
//         max_len
    }
}